suite/src/__tests__/fast/events.test.ts

```diff
@@ -0,0 +1,139 @@
+  }
+}
+async function readEvents(url: string, model: StreamID) {
+  let response = await fetch(url + `/ceramic/subscribe/model/${model.toString()}`)
+  return await response.json()
+}
+
+function sortModelEvents(events: any[]) {
+  return JSON.parse(JSON.stringify(events)).sort((a: any, b: any) => {
+    if (a.eventId > b.eventId) return 1;
+    if (a.eventId < b.eventId) return -1;
+    return 0;
+  })
+}
+
+describe('events', () => {
+  const firstNodeUrl = CeramicUrls[0]
+
+  test(`linear sync on ${firstNodeUrl}`, async () => {
+    const modelID = new StreamID('model', randomCID())
+    let modelEvents = randomEvents(modelID, 10);
+
+    // Write all data to one node before subscribing on the other nodes.
+    // This way the other nodes to a linear download of the data from the first node.
+    await subscribe(firstNodeUrl, modelID)
+    await writeEvents(firstNodeUrl, modelEvents)
+
+    // Now subscribe on the other nodes
+    for (let idx = 1; idx < CeramicUrls.length; idx++) {
+      let url = CeramicUrls[idx]
+      await subscribe(url, modelID)
+    }
+    await delay(3)
+    // Use a sorted expected value for stable tests
+    const sortedModelEvents = sortModelEvents(modelEvents)
+    // Validate each node got the events, including the first node
+    for (let idx in CeramicUrls) {
+      let url = CeramicUrls[idx]
+      let events = await readEvents(url, modelID)
+
+      expect(events).toEqual(sortedModelEvents)
+    }
+  })
+
+  test(`active write sync on ${firstNodeUrl}`, async () => {
+    const modelID = new StreamID('model', randomCID())
+    let modelEvents = randomEvents(modelID, 10);
+    // Subscribe on all nodes then write the data
+    for (let idx in CeramicUrls) {
+      let url = CeramicUrls[idx]
+      await subscribe(url, modelID)
+    }
+    await writeEvents(firstNodeUrl, modelEvents)
+    await delay(3)
+    // Use a sorted expected value for stable tests
+    const sortedModelEvents = sortModelEvents(modelEvents)
+    // Validate each node got the events, including the first node
+    for (let idx in CeramicUrls) {
+      let url = CeramicUrls[idx]
+      let events = await readEvents(url, modelID)
+
+      expect(events).toEqual(sortedModelEvents)
+    }
+  })
+  test(`half and half sync on ${firstNodeUrl}`, async () => {
+    const modelID = new StreamID('model', randomCID())
+    let modelEvents = randomEvents(modelID, 20);
+    // Write half the data before other nodes subscribe
+    await subscribe(firstNodeUrl, modelID)
+    let half = Math.ceil(modelEvents.length / 2);
+    let firstHalf = modelEvents.slice(0, half)
+    let secondHalf = modelEvents.slice(half, modelEvents.length)
+    await writeEvents(firstNodeUrl, firstHalf)
+
+    // Now subscribe on the other nodes
+    for (let idx = 1; idx < CeramicUrls.length; idx++) {
+      let url = CeramicUrls[idx]
+      await subscribe(url, modelID)
+    }
+    // Write the second half of the data
+    await writeEvents(firstNodeUrl, secondHalf)
```

Comment:
The behavior should be symmetric so it shouldn't matter but do we want to write the second half to a different node besides the first one? Just a thought.

Reply:
I like that, I think I'll add an active writes on two nodes test